htdocs/PI/write/PIWriteRequest.php

```diff
@@ -902,14 +904,132 @@ private function updateEndpoint (Site $siteService) {
-        }
+      }
+    }
+  }
+
+  /**
+   * Update a property of an endpoint folloeing a POST requests
+   *
+   * @param  EndpointLocation $endpoint
+   * @param    $endpointPropName Name of property being updated
+   * @param    $endpointPropValue Value of property being updated
+   * @throws \Exception
+   */
+  private function updateEndpointPropPost (\EndpointLocation $endpoint, $endpointPropName, $endpointPropName) {
+    if($endpointPropName == 'monitored') {
+      #Not valid for boolians
+      $this->exceptionWithResponseCode(405,$this->genericExceptionMessages["cantPostABool"]);
+
+    } elseif ($this->serviceService->EndpointPropSet($endpoint, $endpointPropName)) {
+      #POST method must fail if the value is already set
+      $this->exceptionWithResponseCode(409,$this->genericExceptionMessages["propAlreadySet"]);
+
+    } else {
+      $this->updateEndpointProp($endpoint, $endpointPropName, $endpointPropName);
+    }
+  }
+
+  /**
+   * Update a property of an endpoint folloeing a PUT requests
+   *
+   * @param  EndpointLocation $endpoint
+   * @param    $endpointPropName Name of property being updated
+   * @param    $endpointPropValue Value of property being updated
+   * @throws \Exception
+   */
+  private function updateEndpointPropPut (\EndpointLocation $endpoint, $endpointPropName, $endpointPropValue) {
+    #update the values
+    $this->updateEndpointProp($endpoint, $endpointPropName, $endpointPropValue);
+
+  }
+
+  /**
+   * Function to update individual properties of endpoints
+   *
+   * @throws \Exception
+   */
+  Private function updateEndpointProp (\EndpointLocation $endpoint, $endpointPropName, $endpointPropValue) {
+    $this->checkServiceServiceSet();
+
+    $name = $endpoint->getName();
+    $url = $endpoint->getUrl();
+    $interfaceName = $endpoint -> getInterfaceName();
+    $description = $endpoint-> getDescription();
+    $email = $endpoint -> getEMail();
+    $monitored = $endpoint->getMonitored();
+
+    switch($endpointPropName) {
+      case 'name':{
+        $name = $endpointPropValue;
+        break;
+      }
+      case 'url':{
+        $url = $endpointPropValue;
+        break;
+      }
+      case 'interfacename':{
+        $interfaceName = $endpointPropValue;
+        break;
+      }
+      case 'description':{
+        $description = $endpointPropValue;
+        break;
+      }
+      case 'email':{
+        $email = $endpointPropValue;
+        break;
+      }
+      case 'monitored':{
+        $monitored = $endpointPropValue;
+        break;
+      }
+      default:{
+        $this->exceptionWithResponseCode(500,
+          "Internal error: endpoint property name ($endpointPropName) not ".
+          "recognised despite being validated. Please contact a GOCDB administrator and report this error."
+        );
+      }
     }
+
+    // check endpoint's name is unique under the service
+    foreach ( $endpoint->getService()->getEndpointLocations () as $endpointL ) {
```

Comment:
This is done twice?

Reply:
Yep. In the service service it is for the web portal as well. Doing it here ensures an API appropriate error message.

Reply:
(and HTTP response code)